packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -110,25 +137,63 @@ class Navigator extends StatefulComponent {
     assert(onGenerateRoute != null);
   }
 
+  /// The name of the first route to show.
   final String initialRoute;
+
+  /// Called to generate a route for a given [RouteSettings].
   final RouteFactory onGenerateRoute;
+
+  /// Called when [onGenerateRoute] fails to generate a route.
+  ///
+  /// This callback is typically used for error handling. For example, this
+  /// callback might always generate a "not found" page that describes the route
+  /// that wasn't found.
```

Comment:
Also say "A route that doesn't exist might be requested by an Android intent." or something, in case people wonder how to test this particular feature...